chalice/app.py

```diff
@@ -1599,7 +1785,45 @@ def _register_handler(self, handler_type, name, user_handler,
         self._deferred_registrations.append(
             # pylint: disable=protected-access
             lambda app, options: app._register_handler(
-                handler_type, name, user_handler, wrapped_handler,
+                handler_type, name, user_handler, wrapped_handler(app),
                 kwargs, options
             )
         )
+
+    def _get_middleware_handlers(self, event_type):
+        raise RuntimeError("Blueprints are unable to retrieve middleware "
+                           "handlers, must be registered to an app using "
+                           "register_blueprint()")
+
+
+# This class is used to convert any existing/3rd party decorators
+# that work directly on lambda functions with the original signature
+# of (event, context).  By using ConvertToMiddleware you can automatically
+# apply this decorator to every lambda function in a Chalice app.
+# Example:
+#
+# Before:
+#
+# @third_part.decorator
+# def some_lambda_function(event, context): pass
+#
+# Now:
+#
+# app.register_middleware(ConvertToMiddleware(third_party.decorator))
+#
+#
+class ConvertToMiddleware(object):
```

Comment:
Is there a particular reason this is not available in `chalice/__init__.py`? I noticed in the example that gets imported from `chalice.app` instead of `chalice` and I was more used to seeing classes being imported from there.

Reply:
Good catch, this should be in `chalice/__init__.py`.